sentry-android-core/src/main/java/io/sentry/android/core/AppStartState.java

```diff
@@ -0,0 +1,70 @@
+package io.sentry.android.core;
+
+import android.os.SystemClock;
+import java.util.Date;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.TestOnly;
+
+/** AppStartState holds the state of the App Start metric and appStartTime */
+final class AppStartState {
+
+  private static @NotNull AppStartState instance = new AppStartState();
+
+  /** appStart in milliseconds */
+  private @Nullable Long appStart;
+
+  /** appStartEnd in milliseconds */
+  private @Nullable Long appStartEnd;
+
+  /** The type of App start coldStart=true -> Cold start, coldStart=false -> Warm start */
+  private boolean coldStart;
+
+  /** appStart as a Date used in the App's Context */
+  private @Nullable Date appStartTime;
```

Comment:
`m`: Can't we just ditch this variable and use `appStart` instead to create the Date here? This would be less complex and we would use less memory.

Reply:
`appStartTime` is used for `contexts.app.appStartTime`, which was previously created in `DefaultAndroidEventProcessor` but it was not accurate as now, so I'd keep it as it is